registry/auth.go

```diff
@@ -149,8 +149,8 @@ func ResolveAuthConfig(authConfigs map[string]registry.AuthConfig, index *regist
 
 	// Maybe they have a legacy config file, we will iterate the keys converting
 	// them to the new format and testing
-	for registry, ac := range authConfigs {
-		if configKey == ConvertToHostname(registry) {
+	for registryURL, ac := range authConfigs {
```

Comment:
How many of these do we still have in the code? Let's make one big PR that fixes all of them and activate some kind of linter that will fail if this happens

Reply:
Yes, we can enable the `shadow` check from `revive`; I did that on the selinux repository; https://github.com/opencontainers/selinux/blob/847203c9b98dd6d316cacba2c92ecaf065cd8477/.golangci.yml#L23-L36

I may actually have a branch somewhere that enables more linters, but it still needed work